test/integration/dbversion.test.js

```diff
@@ -0,0 +1,45 @@
+describe('User service integration test', () => {
+  const dbVersion = require('../../server/dbversion')
+
+  async function truncateVersionTable () {
+    return dbVersion.umzug.storage.model.destroy({
+      where: {},
+      truncate: true
+    })
+  }
+
+  beforeEach(async () => {
+    await truncateVersionTable()
+  })
+
+  test('dbVersion gets a list of available migrations', () => {
+    expect(dbVersion.availableVersions.length).toBeGreaterThan(0)
+  })
+
+  test('dbVersion validates version in database', async () => {
+    await dbVersion.refreshCurrentDatabaseVersion()
+    expect(dbVersion.currentDatabaseVersion).not.toBe(dbVersion.highestVersion)
+  })
+
+  test('dbVersion reports a list of pending migrations', async () => {
+    const pending = await dbVersion.getPending()
+    expect(pending.length).toBeGreaterThan(0)
+  })
+
+  test('dbVersion reports an issue if there are no database versions stored', async () => {
+    expect.assertions(1)
```

Comment:
I've had problems with async tests caught by including this line, so I'd prefer to leave it in.

Reply:
The `expect.assertions(1)` is redundant and should be removed. 
This can be demonstrated by removing the throw from the code so the test fails. 
The test fails with or without the `expect.assertions(1)` present.

Reply:
I agree with @markysoft on this one. It's good practise to remove any code that isn't necessary so that it doesn't cause confusion for other developers or encourage odd superstitions to bloat code.

It is only necessary to use `expect.assertions()` when assertions could feasibly not be reached during test execution (usually because assertions are placed in a callback function or `catch` block). For example:
```
  test('dbVersion reports an issue if there are no database versions stored', async () => {
    expect.assertions(1)
    try {
      await dbVersion.throwAnyErrors()
    } catch (error) {
      expect(error.message).toMatch('Pending Database migrations exist')
    }
  })
```

Reply:
In fact, writing the above example highlighted to me that the test ought to verify that the correct error has been thrown:
```
  test('dbVersion reports an issue if there are no database versions stored', async () => {
    await expect(dbVersion.throwAnyErrors()).rejects.toThrow('Pending Database migrations exist')
  })
```

The same applies to other tests. We should always verify the values we expect to be returned, thrown or passed to callbacks or other functions as a result of a test.